presidio-anonymizer/presidio_anonymizer/anonymizer_engine.py

```diff
@@ -48,31 +50,55 @@ def anonymize(self, engine_request: AnonymizerRequest) -> str:
             text_to_anonymize = text_builder.get_text_in_position(
                 analyzer_result.start, analyzer_result.end)
 
-            anonymizer_dto = engine_request.get_anonymizer_dto(
-                analyzer_result.entity_type
-            )
+            anonymizer_config = self.__get_anonymizer_by_entity_type(
+                engine_request.get_anonymizers_config(),
+                analyzer_result.entity_type)
+
             self.logger.debug(
                 f"for analyzer result {analyzer_result} received anonymizer "
-                f"{str(anonymizer_dto)}"
+                f"{str(anonymizer_config)}"
             )
 
-            anonymized_text = self.__extract_anonymizer_and_anonymize(anonymizer_dto,
-                                                                      text_to_anonymize)
+            anonymized_text = self.__extract_anonymizer_and_anonymize(
+                analyzer_result.entity_type, anonymizer_config,
+                text_to_anonymize)
             text_builder.replace_text(anonymized_text, analyzer_result.start,
                                       analyzer_result.end)
 
         return text_builder.output_text
 
     def anonymizers(self):
         """Return a list of supported anonymizers."""
-        names = [p for p in self.builtin_anonymizers.keys()]
+        names = [p for p in Anonymizer.get_anonymizers().keys()]
         return names
 
-    def __extract_anonymizer_and_anonymize(self, anonymizer_dto, text_to_anonymize):
-        anonymizer = anonymizer_dto.get("anonymizer")()
+    def __extract_anonymizer_and_anonymize(self, entity_type: str,
```

Comment:
```suggestion
    @staticmethod
    def __extract_anonymizer_and_anonymize(entity_type: str,
```

Reply:
I am using the logger here... I can't do that.